app/javascript/controllers/embed_preview_controller.js

```diff
@@ -0,0 +1,140 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["accentColor", "embedIframe", "embedIframeWrapper", "embedUrl", "embedHtml"]
+
+  static values = {
+    embedUrl: String,
+    embedType: String,
+  }
+
+  connect() {
+    const that = this
+    window.addEventListener(
+      "message",
+      (e) => {
+        that.handlePostMessage(e)
+      },
+      false
+    )
+
+    if (this.hasEmbedIframeTarget && this.hasEmbedUrlValue) {
+      this.embedIframeTarget.setAttribute("src", this.embedUrlValue)
+    }
+  }
+
+  disconnect() {
+    const that = this
+    window.removeEventListener(
+      "message",
+      (e) => {
+        that.handlePostMessage(e)
+      },
+      false
+    )
+  }
+
+  handlePostMessage(e) {
+    if (!/^https?:\/\/play(?:\.staging)?\.prx\.(?:org|tech|test)$/.test(e.origin)) return
```

Comment:
I'm not sure I follow what this code is doing - can you clarify?  (Either here, or in the code itself)